Eshop.Domain/Product.cs

```diff
@@ -1,24 +1,28 @@
-﻿namespace Eshop.Domain
+﻿using System.ComponentModel.DataAnnotations.Schema;
+
+namespace Eshop.Domain
 {
     public class Product
     {
         public Product(int id, string title, string description, decimal price, Category category)
         {
             if (id < 0)
             {
-                throw new ArgumentOutOfRangeException("id");
+                throw new ArgumentOutOfRangeException(nameof(id));
             }
 
             ValidateParameters(title, description, price, category);
 
-            Id = id;
+            Id = id; 
             Title = title;
             Description = description;
             Price = price;
             Category = category;
         }
 
-        public int Id { get; }
+        private Product() { }
```

Comment:
move it to first line - it is a good practice and provide comment why it is needed, becasue from domain driven design it wasn't needed, but now because we are using EF, we need that. So you should provide meaningful comment eg - // For EF persistence purpose

Reply:
added and moved